src/client/mod.rs

```diff
@@ -273,7 +277,7 @@ where C: Connect<Error=io::Error> + Sync + 'static,
     }
 
     fn schedule_pool_timer(&self) {
-        self.pool.spawn_expired_interval(&self.executor);
+        self.pool.spawn_expired_interval(&mut self.executor.clone())
```

Comment:
It's needed because the method itself takes `&self`, so `self.executor` is `&Exec`. Cloning makes it owned and therefore mutable.

The method needs to take `&self` because it is called from `Client::request`, which takes `&self`. Maybe we could fix this up if we moved `schedule_pool_timer` into the constructor for `Client`?

Reply:
Ah OK, right. That `schedule_pool_timer` was introduced as a bit of a hack, and it's still being all hacky. I guess I'd move it to being called from `Checkout::poll`, so that it's only created once the client starts to get used, and there's a `Context` to grab.

Reply:
Moved it into `Checkout::poll` now.